File: 0011-container-with-most-water/0011-container-with-most-water.py

class Solution:
    def maxArea(self, height: List[int]) -> int:
        area = 0
        j = len(height) - 1
        i = 0
        while i < j:
            present = (min(height[i], height[j])) * (j - i)
            if area < present:
                area = present
            if height[i] > height[j]:
                j -= 1
            else:
                i += 1
        return area

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int area=0;
        int present=0 ,j= height.size()-1;
        for(int i=0;i<height.size();i++){
            present=(min(height[i],height[j]))*(j-i);
            if(area<present){
                area=present;
            }
            if(height[i]>height[j]){
                j--;
                i--;
            }
        }
        return area;
        
    }
};